File: 백준/Bronze/1009. 분산처리/분산처리.py
import sys
T = int(sys.stdin.readline())

ans = [[10], [1], [2,4,8,6], [3,9,7,1], [4,6], [5], 
       [6], [7,9,3,1], [8,4,2,6], [9,1]]

for _ in range(T):
    a,b = map(int, sys.stdin.readline().split())
    if a == 10:
        print(10)
    else:
        print(ans[a%10][b%(len(ans[a%10]))-1])

File: 백준/Bronze/1009. 분산처리/분산처리.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();

        for(int i=0; i<t; i++){

            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken())%10;
            int b = Integer.parseInt(st.nextToken());

            ArrayList<Integer> list = new ArrayList<>();
            list.add(a);
            int tmp = a;
            while (true){
                tmp = (tmp * a) % 10;
                if (tmp == a) break;
                else list.add(tmp);
            }

            int ans = list.get((b-1) % list.size());
            if (ans == 0) sb.append(10).append('\n');
            else sb.append(ans).append('\n');
        }

        System.out.println(sb);
    }
}
